src/utils/edge-handlers.js

```diff
@@ -74,11 +74,7 @@ const deployEdgeHandlers = async ({ site, deployId, api, silent, error, warn })
     } catch (e) {
       const text = `Failed deploying Edge Handlers: ${e.message}`
       stopSpinner({ spinner, text, error: true })
-      try {
-        await api.cancelSiteDeploy({ deploy_id: deployId })
-      } catch (e) {
-        warn(`Failed canceling deploy with id ${deployId}: ${e.message}`)
-      }
+      await cancelDeploy({ api, deployId, warn })
```

Comment:
We can throw an error an let the calling function cancel the deploy, but we would need lift the spinner handling too.